Avoid nvcc bug

**Describe the bug**

`nvcc` from CUDA 11.4 misunderstands the name of a `static` member function in a template argument as referring to a non-static instance member:

```
EMPIRE/TPL/vt/src/vt/topos/location/location.impl.h: In member function 'void vt::location::EntityLocationCoord<EntityID>::registerEntity(const EntityID&, const NodeType&, vt::location::LocMsgActionType, const bool&)':
EMPIRE/TPL/vt/src/vt/topos/location/location.impl.h:177:75: error: use of 'this' in a constant expression
  177 |       theMsg()->sendMsg<LocMsgType, updateLocation>(home, msg);
      |                                                                           ^             
EMPIRE/TPL/vt/src/vt/topos/location/location.impl.h: In member function 'void vt::location::EntityLocationCoord<EntityID>::getLocation(const EntityID&, const NodeType&, const NodeActionType&)':
EMPIRE/TPL/vt/src/vt/topos/location/location.impl.h:404:75: error: use of 'this' in a constant expression
  404 |         theMsg()->sendMsg<LocMsgType, getLocationHandler>(home_node, msg);
      |                                                                           ^                 
EMPIRE/TPL/vt/src/vt/topos/location/location.impl.h: In member function 'void vt::location::EntityLocationCoord<EntityID>::sendEagerUpdate(const EntityID&, vt::NodeType, vt::NodeType, vt::NodeType)':
EMPIRE/TPL/vt/src/vt/topos/location/location.impl.h:477:75: error: use of 'this' in a constant expression
  477 |     theMsg()->sendMsg<LocMsgType, recvEagerUpdate>(ask_node, msg);
      |                                                                           ^        
```

Changing those to (e.g.) `&EntityLocationCoord<EntityID>::updateLocation` compiles successfully.

**To Reproduce**

Build EMPIRE with CUDA 11.4. Probably reproducible outside EMPIRE too, but I haven't tried.

--- a/src/vt/topos/location/location.impl.h
+++ b/src/vt/topos/location/location.impl.h
@@ -174,7 +174,7 @@ void EntityLocationCoord<EntityID>::registerEntity(
       );
       msg->setResolvedNode(this_node);
       theMsg()->markAsLocationMessage(msg);
-      theMsg()->sendMsg<LocMsgType, updateLocation>(home, msg);
+      theMsg()->sendMsg<LocMsgType, &EntityLocationCoord<EntityID>::updateLocation>(home, msg);
     }
   }
 }
@@ -424,7 +424,7 @@ void EntityLocationCoord<EntityID>::getLocation(
           this_inst, id, event_id, this_node, home_node
         );
         theMsg()->markAsLocationMessage(msg);
-        theMsg()->sendMsg<LocMsgType, getLocationHandler>(home_node, msg);
+        theMsg()->sendMsg<LocMsgType, &EntityLocationCoord<EntityID>::getLocationHandler>(home_node, msg);
         // save a pending action when information about location arrives
         pending_actions_.emplace(
           std::piecewise_construct,
@@ -497,7 +497,7 @@ void EntityLocationCoord<EntityID>::sendEagerUpdate(
     auto msg = makeMessage<LocMsgType>(
       this_inst, id, ask_node, home_node, deliver_node
     );
-    theMsg()->sendMsg<LocMsgType, recvEagerUpdate>(ask_node, msg);
+    theMsg()->sendMsg<LocMsgType, &EntityLocationCoord<EntityID>::recvEagerUpdate>(ask_node, msg);
   }
 }
 

--- a/src/vt/topos/location/location.impl.h
+++ b/src/vt/topos/location/location.impl.h
@@ -537,7 +537,7 @@ void EntityLocationCoord<EntityID>::routeMsgNode(
 
     auto m = msg;
     // send to the node discovered by the location manager
-    theMsg()->sendMsg<MessageT, routedHandler>(to_node, m);
+    theMsg()->sendMsg<MessageT, &EntityLocationCoord<EntityID>::routedHandler>(to_node, m);
   } else {
     vt_debug_print(
       normal, location,
@@ -863,7 +863,7 @@ template <typename EntityID>
         );
         msg2->setResolvedNode(node);
         theMsg()->markAsLocationMessage(msg2);
-        theMsg()->sendMsg<LocMsgType, updateLocation>(ask_node, msg2);
+        theMsg()->sendMsg<LocMsgType, &EntityLocationCoord<EntityID>::updateLocation>(ask_node, msg2);
       });
       theMsg()->popEpoch(epoch);
       theTerm()->consume(epoch);

